manifests/init.pp

```diff
@@ -69,35 +96,82 @@
 # limitations under the License.
 #
 class openvpn (
-  Boolean $autostart_all                = true,
-  Boolean $manage_service               = true,
-  Hash $client_defaults                 = {},
-  Hash $clients                         = {},
-  Hash $client_specific_config_defaults = {},
-  Hash $client_specific_configs         = {},
-  Hash $revoke_defaults                 = {},
-  Hash $revokes                         = {},
-  Hash $server_defaults                 = {},
-  Hash $servers                         = {},
+  Boolean                              $autostart_all,
+  Boolean                              $manage_service,
+  Stdlib::Absolutepath                 $etc_directory,
+  String[1]                            $group,
+  Boolean                              $link_openssl_cnf,
+  Optional[Stdlib::Absolutepath]       $pam_module_path,
+  Boolean                              $namespecific_rclink,
+  Pattern[/^[23]\.0$/]                 $default_easyrsa_ver,
+  Stdlib::Unixpath                     $easyrsa_source,
+  Variant[String[1], Array[String[1]]] $additional_packages,
+  Optional[Stdlib::Absolutepath]       $ldap_auth_plugin_location,
+
+  Hash                                 $client_defaults                 = {},
+  Hash                                 $clients                         = {},
+  Hash                                 $client_specific_config_defaults = {},
+  Hash                                 $client_specific_configs         = {},
+  Hash                                 $revoke_defaults                 = {},
+  Hash                                 $revokes                         = {},
+  Hash                                 $server_defaults                 = {},
+  Hash                                 $servers                         = {},
 ) {
+  $easyrsa_version = $facts['easyrsa'] ? {
+    undef   => $default_easyrsa_ver,
+    default => $facts['easyrsa'],
+  }
+
+  include openvpn::install
+  include openvpn::config
 
-  class { 'openvpn::params': }
-  -> class { 'openvpn::install': }
-  -> class { 'openvpn::config': }
+  Class['openvpn::install']
+  -> Class['openvpn::config']
   -> Class['openvpn']
 
-  if !$::openvpn::params::systemd {
+  if $facts['service_provider'] != 'systemd' {
     class { 'openvpn::service':
       subscribe => [Class['openvpn::config'], Class['openvpn::install'] ],
     }
+
     if empty($servers) {
```

Comment:
@bastelfreak What about this?

It can be just `if !$servers` or  `if $servers != {}`

Reply:
mhm, good question. We don't have anything for this in our styleguide. On a first guess `empty($servers)` looks as good as `if !$servers`.